hail/src/main/scala/is/hail/expr/ir/Simplify.scala

```diff
@@ -328,6 +328,9 @@ object Simplify {
       TableFilter(t,
         ApplySpecial("&&", Array(p1, p2)))
 
+    case TableFilter(TableKeyBy(child, key, isSorted), p) if canRepartition => TableKeyBy(TableFilter(child, p), key, isSorted)
+    case TableFilter(TableRepartition(child, n, strategy), p) => TableRepartition(TableFilter(child, p), n, strategy)
```

Comment:
Also TableOrderBy?

Reply:
see 3 lines down!

Reply:
that one was already there. Dunno why we forgot these.